zipline/lib/labelarray.py

```diff
@@ -0,0 +1,482 @@
+                    return np.full(self.shape, COMPARE_TO_UNKNOWN, dtype=bool)
+
+                return op(self.as_int_array(), i)
+
+            elif isinstance(other, Number):
+                return NotImplemented
+
+            return op(super(LabelArray, self), other)
+        return method
+
+    __eq__ = _equality_check(eq)
+    __ne__ = _equality_check(ne)
+    del _equality_check
+
+    def view(self, dtype=_NotPassed, type=_NotPassed):
+        if type is _NotPassed and dtype not in (_NotPassed, self.dtype):
+            raise TypeError("Can't view LabelArray as another dtype.")
+
+        # The text signature on ndarray.view makes it look like the default
+        # values for dtype and type are `None`, but passing None explicitly has
+        # different semantics than not passing an arg at all, so we reconstruct
+        # the kwargs dict here to simulate the args not being passed at all.
+        kwargs = {}
+        if dtype is not _NotPassed:
+            kwargs['dtype'] = dtype
+        if type is not _NotPassed:
+            kwargs['type'] = type
+        return super(LabelArray, self).view(**kwargs)
+
+    # In general, we support resizing, slicing, and reshaping methods, but not
+    # numeric methods.
+    SUPPORTED_NDARRAY_METHODS = frozenset([
+        'base',
+        'byteswap',
+        'compress',
+        'copy',
+        'data',
+        'diagonal',
+        'dtype',
+        'flat',
+        'flatten',
+        'item',
+        'itemset',
+        'itemsize',
+        'nbytes',
+        'ndim',
+        'newbyteorder',
+        'ravel',
+        'repeat',
+        'reshape',
+        'resize',
+        'setflags',
+        'shape',
+        'size',
+        'squeeze',
+        'strides',
+        'swapaxes',
+        'take',
+        'trace',
+        'transpose',
+        'view'
+    ])
+    PUBLIC_NDARRAY_METHODS = frozenset([
+        s for s in dir(ndarray) if not s.startswith('_')
+    ])
+
+    # Generate failing wrappers for all unsupported methods.
+    locals().update(
+        {
+            method: _make_unsupported_method(method)
+            for method in PUBLIC_NDARRAY_METHODS - SUPPORTED_NDARRAY_METHODS
+        }
+    )
+
+    def __repr__(self):
+        # This happens if you call a ufunc on a LabelArray that changes the
+        # dtype.  This is generally an indicator that the array has been used
+        # incorrectly, and it means we're no longer valid for anything.
+        if self.dtype != int64_dtype:
+            return "Invalid LabelArray: dtype={}, shape={}".format(
+                self.dtype, self.shape
+            )
+        repr_lines = repr(self.as_string_array()).splitlines()
+        repr_lines[0] = repr_lines[0].replace('array(', 'LabelArray(', 1)
+        repr_lines[-1] = repr_lines[-1].rsplit(',', 1)[0] + ')'
+        # The extra spaces here account for the difference in length between
+        # 'array(' and 'LabelArray('.
+        return '\n     '.join(repr_lines)
```

Comment:
This would look something like:
```
separator = '\n' + ' ' * (len(type(self).__name__) - len('array'))
return separator.join(repr_lines)
```
I'm not sure that's actually much clearer (I would want the comment explaining what's happening either way), so I'd be inclined to just leave this as-is with the comment.